### PR TITLE
[Fabric] Clean up hit testing now that RCTUIView extends RCTPlatformView

### PR DESCRIPTION
**Pull request by lenaic**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

The RCTUIView extends RCTPlatformView now which allows to remove the macOS specific hit testing logic that was added for the implementation of the RCTSurfaceTouchHandler in fabric.

The only macOS specific part that is still required is the implementation of a 
```- (RCTPlatformView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event```
 method to support the iOS-style hit test calls coming from the betterHitTest implementation.

## Changelog

[macOS] [REMOVED] - Removed custom hit test implementation used for macOS fabric
## Test Plan

Tested by running RNTester on macOS with fabric (`RCT_NEW_ARCH_ENABLED=1`) and clicking around the app to check that hit testing works as expected.

https://user-images.githubusercontent.com/151054/236241920-36bdae96-6212-42eb-b0c6-7e918fac6bcd.mov

### CODE REVIEW

File: React/Base/RCTUIKit.h [hunk 1]
```diff
@@ -391,7 +391,6 @@ CGPathRef UIBezierPathCreateCGPathRef(UIBezierPath *path);
 - (void)layoutIfNeeded;
 
 - (void)layoutSubviews;
-- (NSArray<RCTUIView *> *)reactZIndexSortedSubviews; // [macOS]
```
Comment (Saadnajmi) [inline React/Base/RCTUIKit.h:394]: Why is this removed?

**Review by Saadnajmi — APPROVED** [review]
A note: Someone recently pointed out to me that I'm their own custom native components, they need to make sure to flip the y coordinate (as we do in RCTView's hit test) because of macOS's inverted coordinate system. Do we need to do that here too?

File: React/Base/RCTUIKit.h (hunk 1 above)
Comment (lenaic) [reply React/Base/RCTUIKit.h:394]: This was added in 0.68 where RCTUIView did not extend RCTPlatformView. This meant we were not able to use the existing hit test implementation and we needed to re-use the paper hit test implementation that depended on that method.

On fabric, views are always inserted in the right z-index order, so we don't need to re-order views before hit testing.
